test/matchers/to-have-stat-stage.ts

```diff
@@ -0,0 +1,36 @@
+import { Stat, type BattleStat } from "#enums/stat";
+import { isPokemonInstance, receivedStr } from "#test/test-utils/testUtils";
+import type { MatcherState, SyncExpectationResult } from "@vitest/expect";
+
+/**
+ * Matcher to check if a Pokemon has a specific {@linkcode Stat} stage
+ * @param received - The object to check. Should be a {@linkcode Pokemon}.
+ * @param stat - The {@linkcode Stat} to check
+ * @param expectedStage - The expected stage of the {@linkcode stat}
+ * @returns Whether the matcher passed
+ */
+export function toHaveStatStageMatcher(
+  this: MatcherState,
+  received: unknown,
+  stat: BattleStat,
+  expectedStage: number,
+): SyncExpectationResult {
+  if (!isPokemonInstance(received)) {
+    return {
+      pass: this.isNot,
+      message: () => `Expected Pokemon, but got ${receivedStr(received)}!`,
+    };
+  }
+
+  const actualStage = received.getStatStage(stat);
+  const pass = actualStage === expectedStage;
+  const statStr = Stat[stat];
+
+  return {
+    pass,
+    message: () =>
+      pass
+        ? `Expected ${received.name} ${statStr} stage to NOT be ${statStr}, but it is!`
```

Comment:
```suggestion
        ? `Expected ${received.name} ${statStr} stage to NOT be ${actualStage}, but it is!`
```

Reply:
Why, this is the negative/inverted case

Reply:
So that it displays "Expected Pokemon's SPD to NOT be +1" instead of "Expected Pokemon's SPD to NOT be SPD".

Reply:
Oh. But it would probably be `${expectedStage}` then unless my brain is wired wrong right now
```suggestion
        ? `Expected ${received.name} ${statStr} stage to NOT be ${expectedStage}, but it is!`
```

Reply:
That also works (`expectedStage === actualStage` when this test fails).

Reply:
Oh true. But from a logical perspective I’m locked on the expected value. Thanks for helping me resolve this